tests/unit/test_charm.py

```diff
@@ -0,0 +1,153 @@
+    act: Trigger events.
+    assert: Charm in block state.
+    """
+    mock_event = MagicMock()
+    getattr(charm, event)(mock_event)
+
+    assert isinstance(charm.unit.status, ops.BlockedStatus)
+
+
+@pytest.mark.parametrize(
+    "event",
+    [
+        pytest.param("_on_config_changed", id="config_changed"),
+        pytest.param("_on_cache_config_relation_changed", id="config_relation_changed"),
+    ],
+)
+def test_integration_data_not_leader(
+    charm: ContentCacheBackendsConfigCharm, harness: Harness, event: str
+):
+    """
+    arrange: Follow unit with configurations and integration.
+    act: Trigger events.
+    assert: The integration has no data.
+    """
+    harness.set_leader(False)
+    harness.update_config(SAMPLE_CONFIG)
+
+    relation_id = harness.add_relation(
+        CACHE_CONFIG_INTEGRATION_NAME,
+        remote_app="content-cache",
+    )
+    harness.add_relation_unit(relation_id, remote_unit_name="content-cache/0")
+
+    mock_event = MagicMock()
+    getattr(charm, event)(mock_event)
+
+    data = harness.get_relation_data(relation_id, app_or_unit=charm.app.name)
+    assert charm.unit.status == ops.BlockedStatus("Waiting for integration")
+    assert data == {}
+
+
+@pytest.mark.parametrize(
+    "event",
+    [
+        pytest.param("_on_config_changed", id="config_changed"),
+        pytest.param("_on_cache_config_relation_changed", id="config_relation_changed"),
+    ],
+)
+def test_integration_data(charm: ContentCacheBackendsConfigCharm, harness: Harness, event: str):
+    """
+    arrange: Leader unit with configurations and integration.
+    act: Trigger events.
+    assert: The configuration is in the databag.
+    """
+    harness.update_config(SAMPLE_CONFIG)
+
+    relation_id = harness.add_relation(
+        CACHE_CONFIG_INTEGRATION_NAME,
+        remote_app="content-cache",
+    )
+    harness.add_relation_unit(relation_id, remote_unit_name="content-cache/0")
+
+    mock_event = MagicMock()
+    getattr(charm, event)(mock_event)
+
+    data = harness.get_relation_data(relation_id, app_or_unit=charm.app.name)
+    assert charm.unit.status == ops.ActiveStatus()
+    assert data == {
+        "hostname": "example.com",
+        "path": "/",
+        "backends": '["10.10.1.1", "10.1.1.2"]',
+        "protocol": "https",
+    }
+
+
+@pytest.mark.parametrize(
+    "is_leader",
+    [
+        pytest.param(True, id="leader"),
+        pytest.param(False, id="follower"),
+    ],
+)
+def test_integration_removed(
+    harness: Harness, charm: ContentCacheBackendsConfigCharm, is_leader: bool
+):
+    """
+    arrange: Unit with integration.
+    act: Remove integration.
+    assert: Block status
+    """
+    harness.set_leader(is_leader)
+    harness.update_config(SAMPLE_CONFIG)
+
+    relation_id = harness.add_relation(
+        CACHE_CONFIG_INTEGRATION_NAME,
+        remote_app="content-cache",
+    )
+    harness.add_relation_unit(relation_id, remote_unit_name="content-cache/0")
```

Comment:
I'd double check that the status is active before removing the relation, just to make sure that there's been a change of status after removing the relation

Reply:
Done.